votes/0162-new-pr-requirement-and-commit-message-style.md

```diff
@@ -0,0 +1,166 @@
+
+After this proposal, the bot will check whether there is relevant issue number(s) in the PR body, by `Issue Number: ref #{issue-number}` or `Issue Number: close[sd]?|resolve[sd]?|fix(e[sd])? #{issue-number}`. `ref` means associated with an issue but does not close it. There should be no other content in the same line and the check is case-insensitive. Multiple issues should be separated by a comma, like `Issue Number: close #12341, close #12342`. For PRs trying to close issues in a different repository, contributors need to first create an issue in the same repository and use this issue to track. If the pull request body does not provide the required content, the bot will add the `do-not-merge/needs-linked-issue` label to the pull request to prevent it from being merged.
+
+While merging the PR, the bot will extract the content after `Issue Number:` and add them as a new line to the final commit message body. So if a PR `#12345` has a title `pkg: what's changed in this one package` and its body contains `Issue Number: close #12341, close #12342`, after merging, its commit message is
+
+```
+pkg: what's changed in this one package (#12345)
+
+close #12341, close #12342
+```
+
+2. Refactor the final commit message style
+
+We hope to encourage contributors to write good, meaningful commit messages, drop useless messages, make the final commit message compact and clean. The current complex and messy merging rule will be changed to:
+
+The one-line subject rule will not be changed. It's still the pull request title. The final commit message body will be completely refactored. It won't take the commit messages of the commits in the pull request any more. We'll introduce a new element in the PR body:
+
+    ```commit-message
+    any multiple line commit messages that go into
+    the final commit message body
+
+    * fix something 1
+    * fix something 2
+    ``` 
+
+The content in the `commit-message` code block will be extracted and pasted into the final commit message body. The DCO requirement is still needed for each commit in the pull request. But along with the `Co-authored-by` information, they will be grouped and distinguished by the bot and be appended to the final commit message. So after the proposal, for the first sample commit message in the Motivation section, it will be like:
+
+```
+rsmeter: add pubsub datasink  (#11719)
+
+Signed-off-by: Zhenchi <zhongzc_arch@outlook.com>
+Co-authored-by: Wenxuan <hi@breeswish.org>
+```
+
+And if the pull request body contains the issue information `Issue Number: close #12341, close #12342` and the sample `commit-message` code block, the final commit message will be:
+
+```
+rsmeter: add pubsub datasink  (#11719)
+
+any multiple line commit messages that go into
+the final commit message body
+
+* fix something 1
```

Comment:
Does this part still contain some useless information like `"format code", "address comments", "add unit tests"`?

Reply:
I think what it does is that only messages quoted in code block are concatenated into final commit messages. 

Therefore you only see useful messages that are explicitly quoted in commit.

Reply:
Ah, get it. so the author or co-authored need to write `commit message` code block before merging.

Reply:
> Ah, get it. so the author or co-authored need to write commit message code block before merging.

The robot will fetch the commit list and take the commit authors (except PR author) as co-authors, automatically populate to the final commit message, so we don't need to manually fill in the code block.